root/static/scripts/edit/externalLinks.js

```diff
@@ -401,68 +405,86 @@ export class ExternalLink extends React.Component<LinkProps> {
     }
 
     return (
-      <tr>
-        <td>
-          {/* If the URL matches its type or is just empty, display either a
-              favicon or a prompt for a new link as appropriate. */
-            showTypeSelection
-              ? (
-                <LinkTypeSelect
-                  handleTypeChange={props.typeChangeCallback}
-                  type={props.type}
-                >
-                  {props.typeOptions}
-                </LinkTypeSelect>
-              ) : (
+      <React.Fragment>
+        <tr>
+          <td>
+            <label>
+              {props.isOnlyLink
+                ? l('Add link:')
+                : (
+                  props.isLastLink
+                    ? l('Add another link:')
+                    : props.index + 1
```

Comment:
I'm not sure this is useful - it doesn't really say anything about the link, and people can see it's the first in the list or whatever :) If we have nothing good to show, do we need to show anything at all?

Reply:
It is intended to show the index of a link in a list, since originally there're indentation in each table row, so to align with the fieldset above I kept this indentation, we can still discuss and optimize it.

Reply:
Yeah, I understand the reasoning, it just seems kinda... not useful to have "1", "2" etc :) But I also understand it might feel weird to just have nothing at all there as a label. Since the move is intended to eventually allow selecting more than one type here, showing the favicon or something to that effect is probably not ideal, so I'm not sure what would be a good solution that still keeps a label. @mwiencek, @yvanzo: any ideas?

Reply:
I find it useful to see the total number of links in the list. Numbers also help with not getting lost in long lists of external links. And we don’t have anything better to show at the moment.

Reply:
I'm still not a fan of the 1/2/3 labels, I still feel they don't give any benefit over having nothing at all (I don't generally have a reason to care about the number of links). I notice that the favicon is now displayed by the relationship type, but that's actually not that connected to the relationship, it's connected to the URL itself. What about showing the favicon instead of the 1/2/3 label, and just showing the type name under Type? It would look more like on the sidebar, and we can use the generic "no favicon" link for URLs that do not match any of our favicons :) Something like:

![Screenshot from 2021-08-04 13-36-19](https://user-images.githubusercontent.com/1069224/128167162-e2e4b2cf-93ed-4829-9147-4dd03ee38449.png)

Notice that also would show the favicon for Apple Music, which we have but were not showing at all right now :)

Reply:
1. About external favicons:
   1. Icons match either the external website (e.g. Spotify) or the relationship type (e.g. official homepage) - sometimes it is the same (e.g. Wikidata),
   2. most of icons match external websites,
   3. there is currently no icon for many websites (there is a generic icon though) and most of relationship types.

   Thus icon should be displayed at the right place indeed which is most of the time ahead of the link rather than the relationship type.

2. About icons vs numbers:
   1. It can be used in error message as a reference to point to when grouping items together (e.g. to address the below issue <https://github.com/metabrainz/musicbrainz-server/pull/2114#issuecomment-892546511>),
   2. it can be useful to navigate through very long lists of links (e.g. the Beatles have 56 links, test example taken from the forum),
   3. it can help with distinguishing between similar links (e.g. MB artist matching 3 Discogs entries) if provided in addition to icon.

   So numbers can have some benefit and can also be used along with icons which look much nicer indeed. But my main concern about numbers is: It might let editors think numbers are actual data (likewise track numbers) which is not the case. It would probably be less confusing to find another way to refer to other list items from error messages.

P.S. The 1. is a just list a statements, while the 2. is more thinking out loud.

Reply:
<img width="413" alt="screenshot" src="https://user-images.githubusercontent.com/17172063/128629006-959ff895-a973-4d25-bb0e-3d3bb029b60b.png">
Here's the new layout, put favicon before index numbers for better alignment (when there's no favicon).